### PR TITLE
Filter scheme expressions before testing VAT number

### PR DESCRIPTION
**Pull request by proggeler**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC break      | no
| Tests passed? | yes
| Fixed tickets | none
| License       | MIT

Small change in performance. It only test the eligible expressions for the given country, not the complete list.

### CODE REVIEW

File: src/VatNumberValidator.php [hunk 1]
```diff
@@ -90,9 +90,12 @@ public function validate($value, Constraint $constraint)
             return;
         }
 
+        // Use only the expressions for the given country code
+        $schemes = preg_grep($data = '/^\/\^\(' . preg_quote(substr($value, 0, 2), '/') . '/', $this->schemes);
```
Comment (antalaron) [inline src/VatNumberValidator.php]: The `$data` variable is unused, and please make sure the php-cs-fixer check is complete.

**Comment by antalaron** [comment]
Thank you very much, @proggeler !

I've made some performance checks:

Before:
```
$ time vendor/bin/phpunit >/dev/null

real	0m1,696s
user	0m1,582s
sys	0m0,095s
```

After
```
$ time vendor/bin/phpunit >/dev/null

real	0m1,620s
user	0m1,510s
sys	0m0,092s
```

It really improves the code.

I have some request.

**Review by antalaron — CHANGES_REQUESTED** [review]

File: src/VatNumberValidator.php (hunk 1 above)
Comment (proggeler) [reply src/VatNumberValidator.php]: I removed the `$data`, completely missed that one
Can you point out what you mean with the php-cs-fixer check?
I saw some errors on psr2 coding standard when I run phpcs, but it is not within the lines i wrote. Should I apply the psr2 to the code and fix it for you?

Comment (antalaron) [reply src/VatNumberValidator.php]: In the root of the repo, there is a .php_cs.dist file, which contains the rules I usually apply when formatting my code (https://github.com/FriendsOfPHP/PHP-CS-Fixer). Long story short: please remove the spaces around the string concatenating dots, and than it is ready to be merged. Thank you!

Comment (proggeler) [reply src/VatNumberValidator.php]: Thanks for mentioning the php-cs-fixer. I usually depend on my ide. This helps a lot when dealing with different coding standards.

**Review by antalaron — APPROVED** [review]

**Comment by antalaron** [comment]
Thank you, @proggeler !

**Pull request merged**